packages/cli/src/environments.ee/source-control/source-control.service.ee.ts

```diff
@@ -481,6 +487,13 @@ export class SourceControlService {
 	async getStatus(user: User, options: SourceControlGetStatus) {
 		await this.sanityCheck();
 
+		const context = new SourceControlContext(user);
+
+		if (options.direction === 'pull' && !context.accessToAllProjects()) {
```

Comment:
Maybe checking `hasGlobalScope(this.userInternal, 'sourceControl:pull')` would make even more sense, no ?